code/modules/mob/living/carbon/xenomorph/castes/king/king.dm

```diff
@@ -66,6 +112,11 @@
 		if(XENO_HIVE_ADMEME)
 			new_caste_type = /mob/living/carbon/xenomorph/king/admeme
 	var/mob/living/carbon/xenomorph/king/kong = new new_caste_type(src)
+	if(future_king)
+		future_king.mind.transfer_to(kong)
+		future_king.offer_mob()
+		on_king_occupied(kong)
+		return
 	RegisterSignal(kong, COMSIG_MOB_LOGIN , .proc/on_king_occupied)
```

Comment:
I think you can use the login signal in both cases and avoid trying to unregister a null signal if not offered:

```suggestion
	RegisterSignal(kong, COMSIG_MOB_LOGIN , .proc/on_king_occupied)
	if(future_king)
		future_king.mind.transfer_to(kong)
		future_king.offer_mob()
		return
```

Reply:
well, let's try